### PR TITLE
[release-4.6] Add e2e testing of server-side apply for openshift types 

### PR DESCRIPTION
**Pull request by marun**
Backport of #25652 to ensure against regression in 4.6

/cc @sttts @p0lyn0mial

### CODE REVIEW

**Comment by sttts** [comment]
/retest

Clean pick?

File: test/extended/apiserver/apply.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/restmapper"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	exetcd "github.com/openshift/origin/test/extended/etcd"
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+// TODO(marun) Evaluate the CRDs that ocp with (~70 at time of
```
Comment (p0lyn0mial) [inline test/extended/apiserver/apply.go:27]: what are we going to do with this TODO ?

Comment (marun) [reply test/extended/apiserver/apply.go]: Removed. CRD testing will be performed separately.

**Comment by marun** [comment]
> /retest
> 
> Clean pick?

Yes, except for the test annotations which have to be regenerated.

**Comment by marun** [comment]
@p0lyn0mial I'm confused, the test failures indicate that your oauth-apiserver fix isn't working. Didn't that get merged for release-4.6?

**Comment by p0lyn0mial** [comment]
> @p0lyn0mial I'm confused, the test failures indicate that your oauth-apiserver fix isn't working. Didn't that get merged for release-4.6?

It looks like it didn't, still in the queue - https://github.com/openshift/oauth-apiserver/pull/27

**Comment by marun** [comment]
@p0lyn0mial Looks to be merged now!

**Pull request merged**